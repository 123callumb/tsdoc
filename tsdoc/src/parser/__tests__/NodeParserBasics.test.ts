import { TestHelpers } from './TestHelpers';

test('00 Tokenizer simple case', () => {
  TestHelpers.parseAndMatchSnapshot([
    '/**',
    ' * line 1 ', // extra space at end of line
    ' * line 2',
    ' */'
  ].join('\n'));
});

test('01 Tokenizer degenerate cases', () => {
  TestHelpers.parseAndMatchSnapshot('/***/');

  TestHelpers.parseAndMatchSnapshot([
    '/**',
    ' *',
    ' */'
  ].join('\n'));

  TestHelpers.parseAndMatchSnapshot([
    '/**',
    ' ',
    ' ',
    ' */'
  ].join('\n'));

});

test('02 Backslash escapes: positive examples', () => {
  TestHelpers.parseAndMatchSnapshot([
    '/**',
    ' * \\$\\@param',
    ' */'
  ].join('\n'));
});

test('03 Backslash escapes: negative examples', () => {
  TestHelpers.parseAndMatchSnapshot([
    '/**',
    ' * letter: \\A space: \\  end of line: \\',
    ' */'
  ].join('\n'));
});
