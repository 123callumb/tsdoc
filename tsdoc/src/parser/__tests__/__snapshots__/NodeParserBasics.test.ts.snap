// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`00 Tokenizer simple case 1`] = `
Object {
  "buffer": "/**[n] * line 1 [n] * line 2[n] */",
  "lines": Array [
    "line 1",
    "line 2",
  ],
  "verbatimNodes": Array [
    Object {
      "kind": "PlainText",
      "nodeExcerpt": "line 1",
    },
    Object {
      "kind": "SoftBreak",
      "nodeExcerpt": "[n]",
    },
    Object {
      "kind": "PlainText",
      "nodeExcerpt": "line 2",
    },
    Object {
      "kind": "SoftBreak",
      "nodeExcerpt": "[n]",
    },
  ],
}
`;

exports[`01 Tokenizer degenerate cases 1`] = `
Object {
  "buffer": "/***/",
  "lines": Array [],
  "verbatimNodes": Array [],
}
`;

exports[`01 Tokenizer degenerate cases 2`] = `
Object {
  "buffer": "/**[n] *[n] */",
  "lines": Array [
    "",
  ],
  "verbatimNodes": Array [
    Object {
      "kind": "SoftBreak",
      "nodeExcerpt": "[n]",
    },
  ],
}
`;

exports[`01 Tokenizer degenerate cases 3`] = `
Object {
  "buffer": "/**[n] [n] [n] */",
  "lines": Array [
    "",
    "",
  ],
  "verbatimNodes": Array [
    Object {
      "kind": "SoftBreak",
      "nodeExcerpt": "[n]",
    },
    Object {
      "kind": "SoftBreak",
      "nodeExcerpt": "[n]",
    },
  ],
}
`;

exports[`02 Backslash escapes: positive examples 1`] = `
Object {
  "buffer": "/**[n] * [b]$[b]@param[n] */",
  "lines": Array [
    "[b]$[b]@param",
  ],
  "verbatimNodes": Array [
    Object {
      "kind": "EscapedText",
      "nodeExcerpt": "[b]$",
    },
    Object {
      "kind": "EscapedText",
      "nodeExcerpt": "[b]@",
    },
    Object {
      "kind": "PlainText",
      "nodeExcerpt": "param",
    },
    Object {
      "kind": "SoftBreak",
      "nodeExcerpt": "[n]",
    },
  ],
}
`;

exports[`03 Backslash escapes: negative examples 1`] = `
Object {
  "buffer": "/**[n] * letter: [b]A space: [b]  end of line: [b][n] */",
  "lines": Array [
    "letter: [b]A space: [b]  end of line: [b]",
  ],
  "verbatimNodes": Array [
    Object {
      "kind": "PlainText",
      "nodeExcerpt": "letter: ",
    },
    Object {
      "errorLocation": "[b]",
      "errorLocationPrecedingToken": " ",
      "errorMessage": "A backslash can only be used to escape a punctuation character",
      "kind": "ErrorText",
      "nodeExcerpt": "[b]",
    },
    Object {
      "kind": "PlainText",
      "nodeExcerpt": "A space: ",
    },
    Object {
      "errorLocation": "[b]",
      "errorLocationPrecedingToken": " ",
      "errorMessage": "A backslash can only be used to escape a punctuation character",
      "kind": "ErrorText",
      "nodeExcerpt": "[b]",
    },
    Object {
      "kind": "PlainText",
      "nodeExcerpt": "  end of line: ",
    },
    Object {
      "errorLocation": "[b]",
      "errorLocationPrecedingToken": " ",
      "errorMessage": "A backslash can only be used to escape a punctuation character",
      "kind": "ErrorText",
      "nodeExcerpt": "[b]",
    },
    Object {
      "kind": "SoftBreak",
      "nodeExcerpt": "[n]",
    },
  ],
}
`;
